MediaBrowser.Common/Plugins/BasePlugin.cs

```diff
@@ -252,20 +252,23 @@ private TConfigurationType LoadConfiguration()
             }
             catch
             {
-                return (TConfigurationType)Activator.CreateInstance(typeof(TConfigurationType));
+                var config = (TConfigurationType)Activator.CreateInstance(typeof(TConfigurationType));
+                SaveConfiguration(config);
+                return config;
             }
         }
 
         /// <summary>
         /// Saves the current configuration to the file system.
         /// </summary>
-        public virtual void SaveConfiguration()
+        /// <param name="config">Configuration to save.</param>
+        public virtual void SaveConfiguration(TConfigurationType config)
```

Comment:
This function is called directly in some plugins.
Maybe add a function overload?
```
$ grep 'SaveConfiguration()' . -r
./jellyfin-plugin-intros/Jellyfin.Plugin.Intros/IntroManager.cs:            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-iptv/MediaBrowser.Channels.IPTV/Api/ServerApiEndpoints.cs:            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/ServerApiEndpoints.cs:                Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/TraktApi.cs:                            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/TraktApi.cs:                Plugin.Instance.SaveConfiguration();
```